DSI/globus_gridftp_server_iRODS.cpp

```diff
@@ -2067,14 +2076,13 @@ globus_l_gfs_iRODS_recv(
 
         if(transfer_info->pathname == nullptr)
         {
+            std::free(collection);
+            collection = nullptr;
```

Comment:
Consider using `irods::at_scope_exit` to handle these calls.

While the current solution is correct, it doesn't protect the function from future memory leaks. The next person could introduce a memory leak if they are careful.

Reply:
> Consider using `irods::at_scope_exit` to handle these calls.
> 
> While the current solution is correct, it doesn't protect the function from future memory leaks. The next person could introduce a memory leak if they are careful.

I moved all of these into the `at_scope_exit`.